rumqttd/src/server/tls.rs

```diff
@@ -200,39 +200,42 @@ impl TLSAcceptor {
             // Get certificates
             let cert_file = File::open(cert_path);
             let cert_file = cert_file.map_err(|_| Error::ServerCertNotFound(cert_path.clone()))?;
-            let certs = rustls_pemfile::certs(&mut BufReader::new(cert_file));
-            let certs = certs.map_err(|_| Error::InvalidServerCert(cert_path.to_string()))?;
-            let certs = certs
-                .iter()
-                .map(|cert| Certificate(cert.to_owned()))
-                .collect();
+            let certs = rustls_pemfile::certs(&mut BufReader::new(cert_file))
+                .collect::<Result<Vec<_>, _>>()
+                .map_err(|_| Error::InvalidServerCert(cert_path.to_string()))?;
 
             // Get private key
             let key = first_private_key_in_pemfile(key_path)?;
 
             (certs, key)
         };
 
-        let builder = ServerConfig::builder().with_safe_defaults();
+        let builder = ServerConfig::builder();
 
         // client authentication with a CA. CA isn't required otherwise
         #[cfg(feature = "verify-client-cert")]
         let builder = {
             let ca_file = File::open(ca_path);
             let ca_file = ca_file.map_err(|_| Error::CaFileNotFound(ca_path.clone()))?;
             let ca_file = &mut BufReader::new(ca_file);
-            let ca_certs = rustls_pemfile::certs(ca_file)?;
+            let ca_certs = rustls_pemfile::certs(ca_file).collect::<Result<Vec<_>, _>>()?;
             let ca_cert = ca_certs
-                .first()
-                .map(|c| Certificate(c.to_owned()))
+                .into_iter()
+                .next()
                 .ok_or_else(|| Error::InvalidCACert(ca_path.to_string()))?;
 
             let mut store = RootCertStore::empty();
             store
-                .add(&ca_cert)
+                .add(ca_cert)
                 .map_err(|_| Error::InvalidCACert(ca_path.to_string()))?;
 
-            builder.with_client_cert_verifier(Arc::new(AllowAnyAuthenticatedClient::new(store)))
+            // This will only return an error if no trust anchors are provided or invalid CRLs are
+            // provided. We always provide a trust anchor, and don't provide any CRLs, so it should
+            // never return an error and it is safe to unwrap.
```

Comment:
```suggestion
            // This will only return an error if no trust anchors or invalid CRLs are provided.
            // We always provide a trust anchor, and don't provide any CRLs,
            // so it is safe to unwrap.
```